addons/map/functions/fnc_compileFlashlightMenu.sqf

```diff
@@ -18,47 +18,30 @@
 
 #include "script_component.hpp"
 
-params ["_vehicle", "_player", "_parameters"];
+params ["", "_player"];
+
+private["_action", "_actions", "_cfg", "_displayName", "_flashlights", "_icon", "_statement"];
 
 _actions = [];
 _flashlights = [_player] call FUNC(getUnitFlashlights);
 
 //add all carried flashlight menus and on/off submenu actions
 {
-    private ["_cfg", "_displayName"," _icon", "_children", "_parentAction"];
     _cfg = (configFile >> "CfgWeapons" >> _x);
     _displayName = getText (_cfg >> "displayName");
     _icon = getText (_cfg >> "picture");
-
-    _children = {
-        private ["_onAction", "_offAction"];
-        params ["_vehicle", "_player", "_flashlight"];
-
-        _onAction = [
-            (_flashlight + "_On"),
-            "On",
-            "",
-            {[_this select 2] call FUNC(switchFlashlight)},
-            {GVAR(flashlightInUse) != (_this select 2)},
-            {},
-            _flashlight
-        ] call EFUNC(interact_menu,createAction);
-
-        _offAction = [
-            (_flashlight + "_Off"),
-            "Off",
-            "",
-            {[""] call FUNC(switchFlashlight)},
-            {GVAR(flashlightInUse) == (_this select 2)},
-            {},
-            _flashlight
-        ] call EFUNC(interact_menu,createAction);
-
-        [[_onAction, [], _player], [_offAction, [], _player]]
+    _statement = {};
+    
+    _statement = if (GVAR(flashlightInUse) == _x) then {
```

Comment:
This will always return something, so the previous `_statement = {};` initialization isn't required here.